fe/fe-core/src/main/java/com/starrocks/system/Backend.java

```diff
@@ -670,14 +673,19 @@ public boolean handleHbResponse(BackendHbResponse hbResponse) {
             }
 
             heartbeatErrMsg = "";
+            this.heartbeatRetryTimes = 0;
         } else {
-            if (isAlive.compareAndSet(true, false)) {
-                isChanged = true;
-                LOG.info("{} is dead,", this.toString());
-            }
+            if (this.heartbeatRetryTimes < Config.heartbeat_retry_times) {
+                this.heartbeatRetryTimes++;
+            } else {
+                if (isAlive.compareAndSet(true, false)) {
+                    isChanged = true;
+                    LOG.info("{} is dead,", this.toString());
```

Comment:
Maybe adding info about retrying is better.

Reply:
HeartbeatMgr log error response so we know if we are retrying